### PR TITLE
Migrate from TextControl to InputControl to remove margin overrides

### PR DESCRIPTION
**Pull request by brookewp**
## What?

Updated `TextControl` to `InputControl` component for RSS block. 

## Why?

While working on [this project](https://github.com/WordPress/gutenberg/issues/38730) to remove margin overrides, I found that moving this to InputControl required less CSS. It is also the plan to replace TextControl with InputControl per [the storybook docs](https://wordpress.github.io/gutenberg/?path=/docs/components-experimental-inputcontrol--default).

## How?

By adding InputControl and HStack to remove CSS overrides. 

## Testing Instructions

1. Add RSS block to the editor
2. Verify Spacing 
3. Ensure it works the same as before 

| Before | After |
| --- | --- |
| Screenshot 2023-01-14 at 12 38 48 AM | Screenshot 2023-01-14 at 12 39 36 AM |

## Additional Notes

Before the changes, if the screen size was smaller, the input wouldn't match the button size (fixed in this PR): 

Screenshot 2023-01-14 at 12 38 57 AM

However, the button would also move below the input on smaller screen sizes. Should it be below like in trunk?

| Before | After |
| --- | --- |
| Screenshot 2023-01-14 at 12 40 20 AM | Screenshot 2023-01-14 at 12 40 26 AM |

<br />

### CODE REVIEW

File: packages/block-library/src/rss/edit.js [hunk 1]
```diff
@@ -66,17 +67,20 @@ export default function RSSEdit( { attributes, setAttributes } ) {
...
-							value={ feedURL }
-							onChange={ ( value ) =>
-								setAttributes( { feedURL: value } )
-							}
-							className="wp-block-rss__placeholder-input"
-						/>
-						<Button variant="primary" type="submit">
-							{ __( 'Use URL' ) }
-						</Button>
+						<HStack align="baseline">
```
Comment (mirka) [inline packages/block-library/src/rss/edit.js]: Why the align baseline?

**Review by mirka — COMMENTED** [review]
Ah, this is a hard one 😅 (Related: #41961)

To share some context:

- `Placeholder` itself has some very [shaky CSS foundations](https://github.com/WordPress/gutenberg/pull/20676#issuecomment-595669774). So this is a can of worms to start with.
- We will be retiring the `__next36pxDefaultSize` prop (#46741). But that's ok, we can still use it here for the time being as a clean fix. We'll need to revisit these blocks when transitioning to the new size scheme anyway.
- As for the wrapping behavior, we need to think about consistency across the Block Library, which has other blocks with this kind of responsive input & button wrapping (`core/embed` or `core/table`).

tl;dr — I think we should keep the wrapping behavior, so this PR can just remain a clean bug fix and not an entry point into a Placeholder overhaul rabbit hole 🙃 We can still do it cleanly with the HStack by enabling the `wrap` prop, and setting a `min-width: 80%` or something like that to the InputControl.

What do you think?

**Comment by brookewp** [comment]
Thank you for the context! This is a good lesson for me; I need to search GH a bit more thoroughly before I go down these rabbit holes... but I figured if it wasn't something we could do, I would at least learn from it! 😅 

> tl;dr — I think we should keep the wrapping behavior, so this PR can just remain a clean bug fix and not an entry point into a Placeholder overhaul rabbit hole 🙃 We can still do it cleanly with the HStack by enabling the wrap prop, and setting a min-width: 80% or something like that to the InputControl.

>What do you think?

This is a great idea! I guess this whole PR is a bit of an odd in-between step, but hopefully, it is helpful to have the small fix in the meantime.

**Review by mirka — APPROVED** [review]
Thanks, I really appreciate the initiative to fix visual regressions in cleaner ways! 💯
